config/processors/etimologias.processor.ts

```diff
@@ -0,0 +1,39 @@
+import { Response } from 'node-fetch';
```

Comment:
Some improvements for this method:
- we can store the cutoff phrase "Avisanos... " in a variable to avoid repeating it and having to change it in 2 places.
- the cutoff phrase was wrong, so this function was not working (you would've eventually found that out while writing the unit tests, that's why we write them)
- replaced Promise-like syntax with async/await syntax.
- **fixed the encoding problem with non utf-8 response text!!** 🎉 

here's the updated code:
```typescript
import { Response } from 'node-fetch';
import cheerio from 'cheerio';
import { postDocumentToDb } from '../../lib/dbClient';
const FIREBASE_COLLECTION_NAME = 'etymologies';
const CUTOFF_EXPRESSION =
  'Avísanos si tienes más datos o si encuentras \nalgún error.';

export type entry = { word: string; etymology: string };

function toDocument(webTitle: string, webContent: string): entry {
  return {
    word: webTitle,
    etymology: webContent,
  };
}

const getDefinition = ($html: Cheerio) => {
  const definition = $html
    .find('p')
    .text()
    // match any text from <p> elements prior to the cutoff expression
    .match(new RegExp(`([\\s\\S]*.*?)${CUTOFF_EXPRESSION}`, 'g'));
  return definition ? definition[0].replace(CUTOFF_EXPRESSION, '').trim() : '';
};

export default async function (response: Response): Promise<void> {
  const responseText = await response.textConverted();
  const $html = cheerio.load(responseText)('html');

  const title = $html.find('h1').text();
  const text = getDefinition($html);

  postDocumentToDb(FIREBASE_COLLECTION_NAME, toDocument(title, text), title);
}

```